main.go

```diff
@@ -12,6 +24,125 @@ func init() {
 	_ = flag.Set("logtostderr", "true")
 }
 
+func GetCurrentNode(clientset kubernetes.Clientset) (*corev1.Node, error) {
+	nodeName := os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		return nil, errors.New("Environment variable NODE_NAME is not set")
+	}
+
+	node, err := clientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get node: %w", err)
+	}
+
+	return node, nil
+}
+
+func UpdateNodeLabels(
+	clientset *kubernetes.Clientset,
+	instanceData *metadata.InstanceData,
+) error {
+	if instanceData == nil {
+		return fmt.Errorf("instance data received from Linode metadata service is nil")
+	}
+
+	node, err := GetCurrentNode(*clientset)
+	if err != nil {
+		return fmt.Errorf("failed to get the node: %w", err)
+	}
+
+	klog.Infof("Updating node labels with Linode instance data: %v", instanceData)
+
+	node.Labels["linode_label"] = instanceData.Label
+	node.Labels["linode_id"] = strconv.Itoa(instanceData.ID)
+	node.Labels["linode_region"] = instanceData.Region
+	node.Labels["linode_type"] = instanceData.Type
+	node.Labels["linode_host"] = instanceData.HostUUID
+
+	_, err = clientset.CoreV1().Nodes().Update(context.TODO(), node, metav1.UpdateOptions{})
+
+	if err != nil {
+		klog.Errorf("Failed to update labels: %s", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func GetClientset() (*kubernetes.Clientset, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientset, nil
+}
+
+func StartDecorator(client metadata.Client, clientset *kubernetes.Clientset, interval time.Duration) {
+	instanceData, err := client.GetInstance(context.Background())
```

Comment:
you could do this in another issue/pr if you'd like but this would be an easy project to get a real context working, maybe start with a timeout context and set it to something < the interval so you never have your watchers exceeding the interval time?

Reply:
I marked it as TODO so we can do it in the following up PR. Thanks for the suggestion!